src/coord/src/catalog/storage.rs

```diff
@@ -717,17 +717,28 @@ impl Transaction<'_> {
         }
     }
 
-    pub fn remove_compute_instance(&self, name: &str) -> Result<(), Error> {
+    pub fn remove_compute_instance(
+        &self,
+        name: &str,
+        compute_id: ComputeInstanceId,
+    ) -> Result<(), Error> {
         let n = self
             .inner
             .prepare_cached("DELETE FROM compute_instances WHERE name = ?")?
             .execute(params![name])?;
         assert!(n <= 1);
-        if n == 1 {
-            Ok(())
-        } else {
-            Err(SqlCatalogError::UnknownComputeInstance(name.to_owned()).into())
+        if n != 1 {
+            return Err(SqlCatalogError::UnknownComputeInstance(name.to_owned()).into());
         }
+
+        let _ = self
+            .inner
+            .prepare_cached(
+                "DELETE FROM compute_introspection_source_indexes WHERE compute_id = ?",
```

Comment:
To prevent needing the `compute_id`, you should be able to do this delete as a join against `compute_instances` (needing to do this delete first obviously), and then delete the instance.

Reply:
Just updated this, also added a `REFERENCES` constraint to the `compute_introspection_source_indexes` so this relationship is explicit.